docs/configuration.rst

```diff
@@ -971,3 +971,100 @@ This must be used in conjunction with ``encrypted = true`` and must have a custo
 See `Encrypted EBS with a Custom KMS Key <_encrypted_ebs>`. ::
 
     ebs_kms_key_id = xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+
+
+.. _fsx_section:
+
+FSx
+^^^
+Configuration for an attached FSx Lustre file system.
+
+Create a new file system, by omitting ``fsx_fs_id``, or use an existing one by specifying ``fsx_fs_id``. ::
+
+    [fsx fs]
+    shared_dir = /fsx
+    fsx_fs_id = fs-073c3803dca3e28a6
+    storage_capacity = 3600
+    fsx_kms_key_id = 9e8a129c-0e85-459-865b-3a5be974a22b
+    imported_file_chunk_size = 1024
+    export_path = s3://bucket/folder
+    import_path = s3://bucket
+    weekly_maintenance_start_time = 1:00:00
+
+shared_dir
+""""""""""
+**Required** Defines the mount point for the Lustre File system on the master and compute nodes.
+
+The example below will mount the filesystem at /fsx.
+
+Do not use NONE or /NONE as the shared directory.::
+
+    shared_dir = /fsx
+
+fsx_fs_id
+"""""""""
+**Optional** Attach an existing FSx File System.
+
+If this option is specified, all following FSx parameters, such as ``storage_capacity`` are ignored. ::
+
+    fsx_fs_id = fs-073c3803dca3e28a6
+
+storage_capacity
+""""""""""""""""
+**Optional** The storage capacity of the file system in GiB.
+
+The storage capacity has a minimum of 3,600 GiB and is provisioned in increments of 3,600 GiB.
+
+Defaults to 3,600 GiB. ::
+
+    storage_capacity = 3600
+
+fsx_kms_key_id
+""""""""""""""
+**Optional** The ID of your AWS Key Management Service (AWS KMS) key.
+
+This ID is used to encrypt the data in your file system at rest.
+
+This must be used with a custom ``ec2_iam_role``. See `Encrypted EBS with a Custom KMS Key <_encrypted_ebs>`. ::
+
+    fsx_kms_key_id = xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+
+imported_file_chunk_size
+""""""""""""""""""""""""
+**Optional** For files imported from a data repository (using ``import_path``), this value determines the stripe count
+and maximum amount of data per file (in MiB) stored on a single physical disk. The maximum number of disks that a single
+file can be striped across is limited by the total number of disks that make up the file system.
+
+The chunk size default is 1,024 MiB (1 GiB) and can go as high as 512,000 MiB (500 GiB).
+Amazon S3 objects have a maximum size of 5 TB. ::
+
+    imported_file_chunk_size = 1024
+
+export_path
+"""""""""""
+
+**Optional** The S3 path where the root of your file system is exported. The path **must** be in the same S3 bucket as
+the ``import_path`` parameter.
+
+Defaults to ``s3://import-bucket/FSxLustre[creation-timestamp]`` where ``import-bucket`` is the bucket provided in
+``import_path`` parameter. ::
+
+    export_path = s3://bucket/folder
+
+import_path
+"""""""""""
+***Optional** S3 Bucket to load data from into the file system. Also serves as the export bucket. See ``export_path``.
```

Comment:
Might be worth specifying that import happens only at filesystem, hence cluster, creation time